### PR TITLE
Updates Secret Management to Use DynamoDB System

### PR DESCRIPTION
**Pull request by matthewdfuller**
This PR removes the S3 bucket requirement for the setting of secrets for various utilities (Sumo Logic, Datadog, etc.). The actual service units themselves are unchanged as they simply read from etcd as they had previously.

This is _step 1_ in a multi-step process. The next step will be to remove the S3 bucket requirement for actual files (.dockercfg). Step 3 will be to remove all references to S3 bucket and remove it from the infrastructure templates. The final step will be to create another etcd user and/or restrict access to etcd and/or invoke the Dynamo container directly.

Note: this only applies to V3. V2 will not be updated with this system.

### CODE REVIEW

File: v3/setup/flight-director.sh [hunk 1]
```diff
@@ -35,35 +35,4 @@ etcdctl get /FD/FD_ALLOW_MARATHON_UNVERIFIED_TLS false
...
-# 2) Edit this file locally
-# 3) Verify changes with peers
-# 4) Upload edited file back to S3: be-secure-tier/.flight-director
-# 5) Restart flight-director fleet-units via jenkins for changes to take effect
-
-
-while read line || [[ -n "$line" ]]; do
-    etcdctl set $line
-done < ${HOMEDIR}/.flight-director
+etcdctl set /FD/GITHUB_API https://api.github.com
```
Comment (iderdik) [inline v3/setup/flight-director.sh:38]: This should probably not be hardcoded. Most clusters will use internal Github

File: v3/setup/secrets.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+SECRETMAP[FD_GITHUB_ALLOWED_TEAMS]="/FD/GITHUB_ALLOWED_TEAMS"
+SECRETMAP[HUD_GITHUB_CLIENT_ID]="/HUD/client-id"
+SECRETMAP[HUD_GITHUB_CLIENT_SECRET]="/HUD/client-secret"
+SECRETMAP[MARATHON_USERNAME]="/marathon/username"
+SECRETMAP[MARATHON_PASSWORD]="/marathon/password"
+SECRETMAP[SUMOLOGIC_ACCESS_ID]="/sumologic_id"
+SECRETMAP[SUMOLOGIC_SECRET]="/sumologic_secret"
+SECRETMAP[SYSDIG_KEY]="/sysdigkey"
+
+TABLE=`sudo echo $SECRETS_TABLE`
```
Comment (nBerg) [inline v3/setup/secrets.sh:17]: where is SECRETS_TABLE coming from?

File: v3/setup/flight-director.sh (hunk 1 above)
Comment (matthewdfuller) [inline v3/setup/flight-director.sh:38]: True, although this wasn't added or changed in this PR (besides removing the newline at the end). I can make a card.

File: v3/setup/secrets.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+docker pull behance/docker-aws-secrets-downloader:latest
+
+for K in "${!SECRETMAP[@]}"; do
+        PLAINTEXT=`sudo docker run behance/docker-aws-secrets-downloader --table $TABLE --name $K`
+        etcdctl set ${SECRETMAP[$K]} $PLAINTEXT &>/dev/null
+done
```
Comment (iderdik) [inline v3/setup/secrets.sh:24]: nice!

File: v3/setup/secrets.sh (hunk 2 above)
Comment (matthewdfuller) [inline v3/setup/secrets.sh:17]: From this (now merged) PR in infrastructure repo: https://github.com/adobe-platform/infrastructure/commit/f71722bce135eafa17b81b53497bdbcab0891554

File: v3/setup/flight-director.sh (hunk 1 above)
Comment (iderdik) [inline v3/setup/flight-director.sh:38]: OK- currently, all clusters use public git so it hasn't come up

**Comment by iderdik** [comment]
:+1:

**Comment by matthewdfuller** [comment]
I should also mention that this PR makes use of this Docker image from Behance: https://hub.docker.com/r/behance/docker-aws-secrets-downloader/

**Comment by nBerg** [comment]
:+1:

**Pull request merged**